mediapipe/tasks/ios/genai/inference/sources/LlmInference.swift

```diff
@@ -100,123 +81,23 @@ import MediaPipeTasksGenAIC
     try self.init(options: options)
   }
 
-  /// Generates a response based on the input text.
+  /// Creates and returns a session runner that wraps around a new session created by the underlying
+  /// LLM engine.
   ///
   /// - Parameters:
-  ///   - inputText: A `String` that is used to query the LLM.
-  /// - Throws: An error if the LLM's response is invalid.
-  @objc public func generateResponse(inputText: String) throws -> String {
-
-    /// Disallow response generation if another response generation call is already in progress.
-    try shouldContinueWithResponseGeneration()
-
-    let tokens = try llmTaskRunner.predict(inputText: inputText)
-
-    responseGenerationInProgress = false
-
-    guard let humanReadableLlmResponse = LlmInference.humanReadableString(llmResponses: tokens)
-    else {
-      throw GenAiInferenceError.invalidResponse
-    }
-
-    return humanReadableLlmResponse
+  ///   - sessionConfig: The C config of type `LlmSessionConfig` that configures how to execute the 
+  /// model.
+  /// - Returns:
+  ///   - An `LlmSessionRunner` that wraps around a new session.
+  /// - Throws: An error if the underlying engine could not create a session.
+  func createSessionRunner(sessionConfig: LlmSessionConfig) throws -> LlmSessionRunner {
```

Comment:
We do have Session Options in swift. I have not committed the file for the `Session` to keep the current PR size manageable. Swift Options are also declared in that file. 
This method is not public. It has internal visibility and is called by the `init` of the swift `Session`. `Session.init(llmInference: options:)` creates the C struct `LlmSessionConfig` from the swift options provided by the user and then calls `llmInference.createSessionRunner(sessionConfig)`.  

I'll commit the rest of the code once this PR is merged.

Reply:
We can also change this method to `createSessionRunner(options: Session.Options)` and let this method handle creation of the C session config from the options passed in by the Session. I just thought it would be better for the Session to convert its options to the config however it sees fit and not let the LlmInference worry about it.
LMK what you think.

Reply:
> This method is not public. It has internal visibility and is called by the init of the swift Session. Session.init(llmInference: options:) creates the C struct LlmSessionConfig from the swift options provided by the user and then calls llmInference.createSessionRunner(sessionConfig).

Thanks for the explanation!  If you don't mind, can we make all changes in one PR? We have some internal dependency on this swift API, I hope we can make all necessary changes in one PR. Let me know if you have other opinions.

Reply:
I have pushed all the code to this PR. Do have a look